src/global/tracking/tracking.cc

```diff
@@ -98,7 +98,7 @@ void InitPlugin(JApplication *app) {
     ));
 
     app->Add(new JOmniFactoryGeneratorT<ActsToTracks_factory>(
-        "CentralCKFTracks",
+        "CentralCKFTracksUnfiltered",
```

Comment:
We discussed the idea of saving all tracktips (instead of only the first one) for unfiltered tracks. Do you want to modify ActsToTracks module as well for this reason?

Reply:
I agree that saving the information for all trackTips is the right thing to do at the unfiltered level.

To give some context, here is how one example DIS event looks in the CKF before applying the ambiguity solver:

```
In CKF tracking, size of constTracks = 30
For track 0:
tipIndex = 8
lastSeed value = 0
For track 1:
tipIndex = 17
lastSeed value = 1
For track 2:
tipIndex = 21
lastSeed value = 2
For track 3:
tipIndex = 29
lastSeed value = 3
For track 4:
tipIndex = 28
lastSeed value = 3
For track 5:
tipIndex = 42
lastSeed value = 3
For track 6:
tipIndex = 46
lastSeed value = 4
For track 7:
tipIndex = 53
lastSeed value = 5
For track 8:
tipIndex = 74
lastSeed value = 5
For track 9:
tipIndex = 78
lastSeed value = 6
For track 10:
tipIndex = 96
lastSeed value = 6
For track 11:
tipIndex = 100
lastSeed value = 7
For track 12:
tipIndex = 118
lastSeed value = 7
For track 13:
tipIndex = 122
lastSeed value = 8
For track 14:
tipIndex = 143
lastSeed value = 8
For track 15:
tipIndex = 147
lastSeed value = 9
For track 16:
tipIndex = 165
lastSeed value = 9
For track 17:
tipIndex = 169
lastSeed value = 10
For track 18:
tipIndex = 190
lastSeed value = 10
For track 19:
tipIndex = 195
lastSeed value = 11
For track 20:
tipIndex = 194
lastSeed value = 11
For track 21:
tipIndex = 208
lastSeed value = 11
For track 22:
tipIndex = 212
lastSeed value = 12
For track 23:
tipIndex = 222
lastSeed value = 13
For track 24:
tipIndex = 242
lastSeed value = 14
For track 25:
tipIndex = 254
lastSeed value = 15
For track 26:
tipIndex = 262
lastSeed value = 16
For track 27:
tipIndex = 275
lastSeed value = 17
For track 28:
tipIndex = 286
lastSeed value = 18
For track 29:
tipIndex = 294
lastSeed value = 19
In CKF tracking, size of ACTS trajectories: 20
```

So, we have 30 trackTips which are grouped into 20 trajectories (for 20 seeds).

After the ambiguity solver, we have 9 surviving trackTips; and we now group them into 9 trajectories, all of which have a single trackTip:

```
In ambiguity solver, size of selected tracks: 9
For solved track 0, iTrack = 0, track tip index is 8
For solved track 1, iTrack = 1, track tip index is 17
For solved track 2, iTrack = 2, track tip index is 42
For solved track 3, iTrack = 3, track tip index is 74
For solved track 4, iTrack = 4, track tip index is 96
For solved track 5, iTrack = 10, track tip index is 212
For solved track 6, iTrack = 12, track tip index is 242
For solved track 7, iTrack = 13, track tip index is 254
For solved track 8, iTrack = 14, track tip index is 262

Number of output_trajectories: 9
Trajectory number 0 has this many trackTips: 1
trackTip = 8
Trajectory number 1 has this many trackTips: 1
trackTip = 17
Trajectory number 2 has this many trackTips: 1
trackTip = 42
Trajectory number 3 has this many trackTips: 1
trackTip = 74
Trajectory number 4 has this many trackTips: 1
trackTip = 96
Trajectory number 5 has this many trackTips: 1
trackTip = 212
Trajectory number 6 has this many trackTips: 1
trackTip = 242
Trajectory number 7 has this many trackTips: 1
trackTip = 254
Trajectory number 8 has this many trackTips: 1
trackTip = 262
```

Some of the solved trackTips (e.g. trackTip 42) do not correspond to 'trackTip.front()' of one of the unfiltered trajectories (i.e. a trajectory prior to the ambiguity solver). So, since we want to keep all the possible candidates in the unfiltered case for comparison and debugging, we should save all the trackTips for a given trajectory in ActsToTracks.

Reply:
Ok. I made the adjustment. This is what the track phi values looked like previously for the above event:

```
root [2] events->Scan("CentralCKFSeededTrackParameters.phi:CentralCKFSeededTrackParametersUnfiltered.phi")
*        3 *        0 * -2.745423 * -2.745423 *
*        3 *        1 * -2.659463 * -2.659463 *
*        3 *        2 * -0.487999 * -2.973740 *
*        3 *        3 * -0.515227 * -0.547968 *
*        3 *        4 * -0.411117 * -3.030914 *
*        3 *        5 * 0.6830639 * -0.563698 *
*        3 *        6 * 1.2292373 * -0.377481 *
*        3 *        7 * 1.1533687 * -0.367529 *
*        3 *        8 * 1.6516475 * -0.558974 *
*        3 *        9 *           * -0.358227 *
*        3 *       10 *           * -0.551163 *
*        3 *       11 *           * -0.517399 *
Type <CR> to continue or q to quit ==>
*        3 *       12 *           * 0.6830639 *
*        3 *       13 *           * 0.6830969 *
*        3 *       14 *           * 1.2292373 *
*        3 *       15 *           * 1.1533687 *
*        3 *       16 *           * 1.6516475 *
*        3 *       17 *           * 1.1533479 *
*        3 *       18 *           * 1.1533534 *
*        3 *       19 *           * 1.6516684 *
```

You can see that there are 20 unfiltered tracks saved to the PODIO collections and 9 final solved tracks.

With the update, we have the same solved tracks, but 30 unfiltered tracks -- one for each trackTip:

```
*        3 *        0 * -2.745423 * -2.745423 *
*        3 *        1 * -2.659463 * -2.659463 *
*        3 *        2 * -0.487999 * -2.973740 *
*        3 *        3 * -0.515227 * -0.547968 *
*        3 *        4 * -0.411117 * -0.524593 *
*        3 *        5 * 0.6830639 * -0.487999 *
*        3 *        6 * 1.2292373 * -3.030914 *
*        3 *        7 * 1.1533687 * -0.563698 *
*        3 *        8 * 1.6516475 * -0.515227 *
*        3 *        9 *           * -0.377481 *
*        3 *       10 *           * -0.411117 *
*        3 *       11 *           * -0.367529 *
Type <CR> to continue or q to quit ==>
*        3 *       12 *           * -0.411152 *
*        3 *       13 *           * -0.558974 *
*        3 *       14 *           * -0.515152 *
*        3 *       15 *           * -0.358227 *
*        3 *       16 *           * -0.411137 *
*        3 *       17 *           * -0.551163 *
*        3 *       18 *           * -0.514914 *
*        3 *       19 *           * -0.517399 *
*        3 *       20 *           * -0.481105 *
*        3 *       21 *           * -0.483472 *
*        3 *       22 *           * 0.6830639 *
*        3 *       23 *           * 0.6830969 *
*        3 *       24 *           * 1.2292373 *
*        3 *       25 *           * 1.1533687 *
*        3 *       26 *           * 1.6516475 *
*        3 *       27 *           * 1.1533479 *
*        3 *       28 *           * 1.1533534 *
*        3 *       29 *           * 1.6516684 *
```